api/models/resourcecleanupstatus.go

```diff
@@ -0,0 +1,98 @@
+package models
+
+import (
+	"fmt"
+	"time"
+)
+
+var resourceCleanupStatusStateTransitions = map[ResourceCleanupStatusState][]ResourceCleanupStatusState{
+	ResourceCleanupStatusStatePending: {
+		ResourceCleanupStatusStateSkipped,
+		ResourceCleanupStatusStateFailed,
+		ResourceCleanupStatusStateDone,
+	},
+}
+
+var resourceCleanupStatusReasonMapping = map[ResourceCleanupStatusState][]ResourceCleanupStatusReason{
+	ResourceCleanupStatusStatePending: {
+		ResourceCleanupStatusReasonAssetScanCreated,
+	},
+	ResourceCleanupStatusStateSkipped: {
+		ResourceCleanupStatusReasonDeletePolicy,
+	},
+	ResourceCleanupStatusStateFailed: {
+		ResourceCleanupStatusReasonProviderError,
+		ResourceCleanupStatusReasonInternalError,
+	},
+	ResourceCleanupStatusStateDone: {
+		ResourceCleanupStatusReasonSuccess,
+	},
+}
+
+func NewResourceCleanupStatus(s ResourceCleanupStatusState, r ResourceCleanupStatusReason, m *string) *ResourceCleanupStatus {
+	return &ResourceCleanupStatus{
+		State:              s,
+		Reason:             r,
+		Message:            m,
+		LastTransitionTime: time.Now(),
+	}
+}
+
+func NewResourceCleanupStatusWithDefaults() *ResourceCleanupStatus {
+	return NewResourceCleanupStatus(ResourceCleanupStatusStatePending, ResourceCleanupStatusReasonAssetScanCreated, nil)
+}
+
+func (rs *ResourceCleanupStatus) Equals(r *ResourceCleanupStatus) bool {
```

Comment:
Can you please add unit tests for the following functions?

Reply:
I wouldn't do that for the following 2 reasons:
- we have now `e2e` tests in place which tests this function as well, and I'd feel an unit test for this would be redundant
- there are currently no unit tests written in the `api` package and I'm not sure if we want to introduce it now with all it's dependencies; I think it'd need a discussion whether we want to go down that path